From ef77948760cbac4df81b102e661f49f2d1782cc8 Mon Sep 17 00:00:00 2001
From: Bariccatti <bariccatti@gmail.com>
Date: Sat, 25 Apr 2026 14:37:52 -0300
Subject: [PATCH] Remove system_ext_specific from all Samsung vendor HIDL stubs

The Samsung vendor HIDL stubs have system_ext_specific: true which
makes the generated -java modules non-generic in Soong. This causes
the GSI build to fail because aosp_shared_system_image enforces
generic-only modules.

Remove system_ext_specific from all Samsung radio (2.0, 2.1, 2.2)
and sysinput (1.0, 1.1, 1.2) HIDL stub versions to fix the full
transitive dependency chain.
---
 vendor/samsung/hardware/radio/2.0/Android.bp    | 1 -
 vendor/samsung/hardware/radio/2.1/Android.bp    | 1 -
 vendor/samsung/hardware/radio/2.2/Android.bp    | 1 -
 vendor/samsung/hardware/sysinput/1.0/Android.bp | 1 -
 vendor/samsung/hardware/sysinput/1.1/Android.bp | 1 -
 vendor/samsung/hardware/sysinput/1.2/Android.bp | 1 -
 6 files changed, 6 deletions(-)

diff --git a/vendor/samsung/hardware/radio/2.0/Android.bp b/vendor/samsung/hardware/radio/2.0/Android.bp
index d03a72b..375e041 100644
--- a/vendor/samsung/hardware/radio/2.0/Android.bp
+++ b/vendor/samsung/hardware/radio/2.0/Android.bp
@@ -3,7 +3,6 @@
 hidl_interface {
     name: "vendor.samsung.hardware.radio@2.0",
     root: "vendor",
-    system_ext_specific: true,
     srcs: [
         "ISehRadio.hal",
         "ISehRadioIndication.hal",
diff --git a/vendor/samsung/hardware/radio/2.1/Android.bp b/vendor/samsung/hardware/radio/2.1/Android.bp
index 56ec166..ba2196a 100644
--- a/vendor/samsung/hardware/radio/2.1/Android.bp
+++ b/vendor/samsung/hardware/radio/2.1/Android.bp
@@ -3,7 +3,6 @@
 hidl_interface {
     name: "vendor.samsung.hardware.radio@2.1",
     root: "vendor",
-    system_ext_specific: true,
     srcs: [
         "ISehRadio.hal",
     ],
diff --git a/vendor/samsung/hardware/radio/2.2/Android.bp b/vendor/samsung/hardware/radio/2.2/Android.bp
index d899222..78b95c5 100644
--- a/vendor/samsung/hardware/radio/2.2/Android.bp
+++ b/vendor/samsung/hardware/radio/2.2/Android.bp
@@ -3,7 +3,6 @@
 hidl_interface {
     name: "vendor.samsung.hardware.radio@2.2",
     root: "vendor",
-    system_ext_specific: true,
     srcs: [
         "types.hal",
         "ISehRadio.hal",
diff --git a/vendor/samsung/hardware/sysinput/1.0/Android.bp b/vendor/samsung/hardware/sysinput/1.0/Android.bp
index 83cde29..5e34890 100644
--- a/vendor/samsung/hardware/sysinput/1.0/Android.bp
+++ b/vendor/samsung/hardware/sysinput/1.0/Android.bp
@@ -3,7 +3,6 @@
 hidl_interface {
     name: "vendor.samsung.hardware.sysinput@1.0",
     root: "vendor",
-    system_ext_specific: true,
     srcs: [
         "ISehSysInputDev.hal",
     ],
diff --git a/vendor/samsung/hardware/sysinput/1.1/Android.bp b/vendor/samsung/hardware/sysinput/1.1/Android.bp
index fef5fb0..daa1a1c 100644
--- a/vendor/samsung/hardware/sysinput/1.1/Android.bp
+++ b/vendor/samsung/hardware/sysinput/1.1/Android.bp
@@ -3,7 +3,6 @@
 hidl_interface {
     name: "vendor.samsung.hardware.sysinput@1.1",
     root: "vendor",
-    system_ext_specific: true,
     srcs: [
         "ISehSysInputDev.hal",
     ],
diff --git a/vendor/samsung/hardware/sysinput/1.2/Android.bp b/vendor/samsung/hardware/sysinput/1.2/Android.bp
index 9d9f486..d1a0aa4 100644
--- a/vendor/samsung/hardware/sysinput/1.2/Android.bp
+++ b/vendor/samsung/hardware/sysinput/1.2/Android.bp
@@ -3,7 +3,6 @@
 hidl_interface {
     name: "vendor.samsung.hardware.sysinput@1.2",
     root: "vendor",
-    system_ext_specific: true,
     srcs: [
         "ISehSysInputDev.hal",
     ],
-- 
2.53.0.windows.3

